UnicornEngine/video/include/unicorn/video/vulkan/ShaderProgram.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_SHADER_PROGRAM_HPP
+#define UNICORN_VIDEO_SHADER_PROGRAM_HPP
+
+#include <vulkan/vulkan.hpp>
+#include <array>
+
+namespace unicorn
+{
+namespace video
+{
+namespace vulkan
+{
+/**
+ * @brief Abstraction for shader program, which renderer will use for rendering meshes
+ */
+class ShaderProgram
+{
+public:
+    /**
+     * @brief Constructor
+     * @param m_device Which device we will you to allocate memory
+     * @param vertShaderPath Path to vertex shader path
+     * @param fragShaderPath Path to fragment shader 
+     */
+    ShaderProgram(vk::Device m_device, const std::string& vertShaderPath, const std::string& fragShaderPath);
+    ShaderProgram() = delete;
```

Comment:
You don't have to delete default constructor if you've specified some other constructor